x/stakeibc/keeper/unbonding_records.go

```diff
@@ -86,14 +90,84 @@ func (k Keeper) GetTotalUnbondAmountAndRecordsIds(ctx sdk.Context, chainId strin
 		k.Logger(ctx).Info(utils.LogWithHostZone(chainId, "Epoch %d - Status: %s, Amount: %v",
 			epochUnbonding.EpochNumber, hostZoneRecord.Status, hostZoneRecord.NativeTokenAmount))
 
-		// We'll unbond all records that have status UNBONDING_QUEUE and have an amount g.t. zero
-		if hostZoneRecord.Status == recordstypes.HostZoneUnbonding_UNBONDING_QUEUE && hostZoneRecord.NativeTokenAmount.GT(sdkmath.ZeroInt()) {
-			totalUnbonded = totalUnbonded.Add(hostZoneRecord.NativeTokenAmount)
-			unbondingRecordIds = append(unbondingRecordIds, epochUnbonding.EpochNumber)
-			k.Logger(ctx).Info(utils.LogWithHostZone(chainId, "  %v%s included in total unbonding", hostZoneRecord.NativeTokenAmount, hostZoneRecord.Denom))
+		if hostZoneRecord.ShouldInitiateUnbonding() {
+			epochNumbers = append(epochNumbers, epochUnbonding.EpochNumber)
+			hostZoneUnbondingMap[epochUnbonding.EpochNumber] = *hostZoneRecord
 		}
 	}
-	return totalUnbonded, unbondingRecordIds
+	return epochNumbers, hostZoneUnbondingMap
+}
+
+// Gets the total unbonded amount for a host zone by looping through the epoch unbonding records
+// Also returns the epoch unbonding record ids
+func (k Keeper) GetTotalUnbondAmount(ctx sdk.Context, hostZoneUnbondingRecords map[uint64]recordstypes.HostZoneUnbonding) (totalUnbonded sdkmath.Int) {
+	totalUnbonded = sdk.ZeroInt()
+	for _, hostZoneRecord := range hostZoneUnbondingRecords {
+		totalUnbonded = totalUnbonded.Add(hostZoneRecord.NativeTokenAmount)
+	}
+	return totalUnbonded
+}
+
+// Given a list of redemption record IDs and a redemption rate, sets the native token amount on each record and returns the total
+// The native amount is calculated using the redemption rate. If a zero redemption rate is passed, set the amount to zero
+func (k Keeper) RefreshUserRedemptionRecordNativeAmounts(
+	ctx sdk.Context,
+	chainId string,
+	redemptionRecordIds []string,
+	redemptionRate sdk.Dec,
+) (totalNativeAmount sdkmath.Int) {
+	// Loop and set the native amount for each record, keeping track fo the total
+	totalNativeAmount = sdkmath.ZeroInt()
+	for _, userRedemptionRecordId := range redemptionRecordIds {
+		userRedemptionRecord, found := k.RecordsKeeper.GetUserRedemptionRecord(ctx, userRedemptionRecordId)
+		if !found {
+			k.Logger(ctx).Error(utils.LogWithHostZone(chainId, "No user redemption record found for id %s", userRedemptionRecordId))
+			continue
+		}
+
+		// Calculate the number of native tokens using the redemption rate
+		nativeAmount := sdk.NewDecFromInt(userRedemptionRecord.StTokenAmount).Mul(redemptionRate).RoundInt()
+		totalNativeAmount = totalNativeAmount.Add(nativeAmount)
+
+		// Set the native amount on the record
+		userRedemptionRecord.Amount = nativeAmount
+		k.RecordsKeeper.SetUserRedemptionRecord(ctx, userRedemptionRecord)
+	}
+	return totalNativeAmount
+}
+
+// Sets the native token amount unbonded on the host zone unbonding record and the associated user redemption records
+func (k Keeper) RefreshHostZoneUnbondingNativeTokenAmounts(
```

Comment:
```suggestion
func (k Keeper) RefreshHostZoneUnbondingNativeTokenAmount(
```

Since there's just one variable we're refreshing on the HZU

Reply:
done